### PR TITLE
``Node`` updates transferred to ``Manager`` and relay to the front-end. Closes #92

### LINKED ISSUE #92: Node updates transferred to Manager and relay to the front-end

### PR DESCRIPTION
**Pull request by edavalosanaya**
- Added test for WS communications to the front-end by using a mock ``Client``.
- Whenever the ``Manager`` receives an ``Node`` state update from a ``Worker``, it broadcast this to all WS clients.

### CODE REVIEW

File: chimerapy/manager.py [hunk 1]
```diff
@@ -172,6 +173,10 @@ async def update_nodes_status(self, request: web.Request):
         logger.debug(f"{self}: Nodes status update to: {self.state.workers}")
 
         # Relay information to front-end
+        await self.server.async_broadcast(
+            signal=MANAGER_MESSAGE.NODE_STATUS_UPDATE,
+            data=self.state.to_dict(),
+        )
```
Comment (umesh-timalsina) [inline chimerapy/manager.py:179]: After https://github.com/oele-isis-vanderbilt/ChimeraPy/pull/115, is merged, I think this belongs to `API`?

**Review by umesh-timalsina — COMMENTED** [review]
A single comment. I think this is good to go.

File: chimerapy/manager.py (hunk 1 above)
Comment (edavalosanaya) [reply chimerapy/manager.py:179]: That is a great point. Okay, will add commit to refactor the code into ``API``.

**Comment by edavalosanaya** [comment]
Rebased branch to match the latest version of `main`, including the commits from PR #115.

File: chimerapy/manager.py (hunk 1 above)
Comment (edavalosanaya) [reply chimerapy/manager.py]: Added to the latest commit 36802b1

**Review by umesh-timalsina — APPROVED** [review]

**Comment by umesh-timalsina** [comment]
LGTM!!

**Pull request merged**